### PR TITLE
[media] Use a single splash image for all platforms

### PR DESCRIPTION
**Pull request by enen92**
## Description
We support both splash.png and splash.jpg but have always shipped the splash in jpeg format. With the recent addition of the webOS platform, Kodi started to ship an additional png splash which is used (I guess) by the OS on the application startup.
As far as I am aware the splash image we ship is only used by Kodi itself so...let's make our life easier and just switch to a single .png splash.
The PR also inverts the order of preference for the images so png is found first.

## Motivation and context
Help @fuzzard live longer

## How has this been tested?
Runtime tested on macOS

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by sundermann — APPROVED** [review]

**Comment by fuzzard** [comment]
Its not the core stuff thats the issue. You'll need to touch base with sam, as all the media art is jpg currently. See if he can convert them all to png, which probably means adding extra steps for compression or something. I dont know what his workflows are like. I just pull a file from a folder.

**Comment by sundermann** [comment]
`pngcrush -brute -ow` is probably a good idea after exporting the artwork from the graphics program.

**Comment by enen92** [comment]
> Its not the core stuff thats the issue. You'll need to touch base with sam, as all the media art is jpg currently. See if he can convert them all to png, which probably means adding extra steps for compression or something. I dont know what his workflows are like. I just pull a file from a folder.

I know it's not the issue but it doesn't really make sense to duplicate splash images anyway, just use the common denominator

**Comment by enen92** [comment]
@kambala-decapitator it looks like splash.jpg is also used in iOS. Mind to check it this causes any troubles?

**Comment by kambala-decapitator** [comment]
> @kambala-decapitator it looks like splash.jpg is also used in iOS. Mind to check it this causes any troubles?

png should be good on both iOS and tvOS. Will make runtime test today.

**Review by kambala-decapitator — APPROVED** [review]
works fine on iOS, but for some reason the system launch image isn't shown on tvOS (17). Anyway, I wouldn't call it a blocker, will create a dedicated issue for investigation.

**Comment by enen92** [comment]
> works fine on iOS, but for some reason the system launch image isn't shown on tvOS (17). Anyway, I wouldn't call it a blocker, will create a dedicated issue for investigation.

This is not urgent, please take your time to investigate. Was it working before with the splash.jpg?

**Comment by kambala-decapitator** [comment]
> Was it working before with the splash.jpg?

yes, in the latest nightly it's fine

File: xbmc/platform/darwin/tvos/Assets.xcassets/LaunchImage.launchimage/Contents.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     {
       "orientation" : "landscape",
       "idiom" : "tv",
-      "filename" : "splash.jpg",
+      "filename" : "splash.png",
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/tvos/Assets.xcassets/LaunchImage.launchimage/Contents.json:6]: line 14 in this file must also become png

please also change the symlink in https://github.com/xbmc/xbmc/tree/master/xbmc/platform/darwin/tvos/Assets.xcassets/LaunchImage.launchimage

**Comment by da-anda** [comment]
I'd honestly go for JPG. Back in the days when we had 640x??? centered splash logos, PNG was the better option. But now that we have visually complex 4K artwork, PNG certainly is no longer the file format to use (also, 4K PNG files likely are huge if they not only contain simple artwork)

**Comment by sundermann** [comment]
The issue is that webOS requires a splash png screen to conform with LG's app store guidelines

**Comment by da-anda** [comment]
how is the webOS splash screen used? Honestly, I think this "splash" should NOT be our internal version specific splash screen but a simplified logo, just like all the other app store artworks. So keep our splash jpg and use a simplified logo for the LG app store. That would be my take on it

**Comment by enen92** [comment]
> how is the webOS splash screen used? Honestly, I think this "splash" should NOT be our internal version specific splash screen but a simplified logo version, just like all the other app store artworks. So keep our splash jpg and use a simplified logo variant for the LG app store. Would that work for webOS or would the startup process look bad then?
> 
> This at least would be my take on it

Yeah I somehow agree. Having played with windows on uwp recently it seems to also display a splash image on launch but only with a simple Kodi logo and brand. Maybe we can reuse that?

**Comment by enen92** [comment]
Looks like in UWP this is the one used: https://github.com/xbmc/xbmc/blob/master/tools/windows/packaging/uwp/media/SplashScreen.scale-200.png

@sundermann would this work for webos?

**Comment by da-anda** [comment]
> Looks like in UWP this is the one used: https://github.com/xbmc/xbmc/blob/master/tools/windows/packaging/uwp/media/SplashScreen.scale-200.png
> 
> @sundermann would this work for webos?

if it is used (which would work for me), either a symlink or copy of this file should be used and moved to https://github.com/xbmc/xbmc/tree/master/tools/webOS/packaging (maybe even move all the images to a `media` subfolder, like it is the case with other platforms)

**Comment by sundermann** [comment]
It is used to display an image while the app is loading in the background. I think this is similar to what happens on Android.

@enen92: That image is too small it needs to be at least 1920x1080.

**Comment by enen92** [comment]
Superseeded by https://github.com/xbmc/xbmc/pull/24137

**Pull request closed without merging**